app/code/Magento/Theme/Block/Html/Title.php

```diff
@@ -79,7 +79,7 @@ public function getPageHeading()
     {
         $pageTitle = !empty($this->pageTitle) ? $this->pageTitle : $this->pageConfig->getTitle()->getShortHeading();
 
-        return $this->shouldTranslateTitle() ? __($pageTitle) : $pageTitle;
+        return $this->shouldTranslateTitle() ? __($pageTitle) : (string)$pageTitle;
```

Comment:
Can we fix this elsewhere? The `__()` function also requires its argument to be a string, so the current suggestion isn't necessarily enough to fix the problem advertised.

Does the problem value come from `$this->pageTitle` or `...->getShortHeading()`?
Perhaps we can improve the type information in the `setPageTitle()` method in this class, or fix the return value of `...->getShortHeading()` if that's the source of the "bad" value.

Reply:
`__()` does not require a string: `gettype(__(null)->render())` = `string` so the given fix works perfectly for us.
Both `pageTitle` and `getShortHeading` are not typed and will not be in the near future.

Reply:
Perhaps the code has changed in the *years* since my comment. It looks like the Phrase constructor now does its own cast to a string for its argument.

https://github.com/magento/magento2/blob/869d8b61fc14c0c27bb9f2d8bbd23b8b6cbb4d66/lib/internal/Magento/Framework/Phrase/__.php#L16-L24
https://github.com/magento/magento2/blob/869d8b61fc14c0c27bb9f2d8bbd23b8b6cbb4d66/lib/internal/Magento/Framework/Phrase.php#L67-L72

However, my comment regarding fixing the definition of the `$pageTitle` variable is still relevant here. We should not need to cast the value on this line.

Reply:
`pageTitle` is not typed and will probably never be. So it could be anything, especially when dealing with 3rd party code :-)
`getPageHeading` is not typed as well, so about your point fixing it somewhere else, I don't quite get where exactly.
We could use `trim((string)$block->getPageHeading()` in `title.phtml`, if you prefer that ...

We won't have the issue anywhere else, the only problem is that `trim()` _expects_ a `string` as input, while Magento is not and probably will never be able to provide strict types in 2.x.

Reply:
From a quick search just now, it looks like the property is typed (via a comment) as a string, so the root cause of the bug you're witnessing seems to be that `getShortHeading()` returns `mixed` and not `string`. Can we fix this in that method instead?

https://github.com/magento/magento2/blob/869d8b61fc14c0c27bb9f2d8bbd23b8b6cbb4d66/lib/internal/Magento/Framework/View/Page/Title.php#L88

Reply:
`$this->pageTitle` can be `null` as well.
See for example `\Magento\Catalog\Helper\Product\View::preparePageMetadata` where `$pageMainTitle->setPageTitle($product->getName())` is used, while `$product` could possibly have a `null` name if imported via a wrongly formatted CSV-file by the customer (a case we experienced sometimes already)

I think there are too many possibilities all which only could be resolved by typing all corresponding methods and properties, which imo has a too big impact on stability, thinking about the thousands of extensions that are out there that may probably stop working after we enforce a `string` type on e.g. `setPageTitle`. The casting to string was the least risky option in my head, but feel free to adjust the PR if you feel other.